### DISCUSSION TITLE
How to limit authentication methods server side?

### DISCUSSION THREAD

**Discussion by Immortalin** [pocketbase/pocketbase#5224] [Q&A] [answered]
How do we limit authentication to Oauth on the server side? I want to prevent malicious users from being able to call the create user API directly with email and password auth.

**Comment by ganigeorgiev** [comment, marked as answer]
If you want to restrict users creation to only from OAuth2 then you can try specifying the [`@request.context` field](https://pocketbase.io/docs/api-rules-and-filters/#filters-syntax) in your auth collection's Create API rule:

```js
@request.context = "oauth2"
```

Additionally, you can also restrict password auth for existing accounts from the _"Collection edit panel > Options"_ tab.
